homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -227,17 +222,14 @@
         entity_id_list = service.data.get(ATTR_ENTITY_ID)
         climate_profile_index = service.data.get(ATTR_CLIMATE_PROFILE_INDEX) - 1
 
-        for hapid in hass.data[DOMAIN]:
-            hap = hass.data[DOMAIN][hapid]
-            home = hap.home
-
-            if entity_id_list:
+        for hap in hass.data[DOMAIN].values():
+            if entity_id_list and entity_id_list != "all":
```

Comment:
The first part shouldn't be needed since the option is required now.

Reply:
done